dashboard/src/actions/overviewActions.js

```diff
@@ -428,42 +426,43 @@ export const getKeySummary = async (dispatch, getState) => {
     });
     if (response.status === 200) {
       if (response.data.keys) {
-        dispatch({
-          type: TYPES.GET_KEY_SUMMARY,
-          payload: response.data,
-        });
-        dispatch(parseKeySummaryforTree());
+        dispatch(parseKeySummaryforTree(response.data.keys));
       }
     }
   } catch (error) {
     dispatch(showToast(DANGER, ERROR_MSG));
   }
 };
-
-export const parseKeySummaryforTree = () => (dispatch, getState) => {
+/**
+ * Function to parse keySummary for the Tree View with checkboxes
+ * @function
+ * @param {Object} keySummary - dataset key summary
+ * @return {Function} - dispatch the action and update the state
+ */
+export const parseKeySummaryforTree = (keySummary) => (dispatch, getState) => {
   const parsedData = [];
-  const keySummary = getState().overview.keySummary.keys;
+
   const optionRoles = Object.keys(keySummary);
+
   const checkedItems = getState().overview.checkedItems;
 
   for (const item of optionRoles) {
     const dataObj = { title: item, options: [] };
     for (const key in keySummary[item]) {
-      if (!isNonEssentialKey(key)) {
+      if (!isServerInternal(`${item}${CONSTANTS.KEYS_JOIN_BY}${key}`)) {
```

Comment:
If `KEYS_JOIN_BY` is `*`, then we try to check `server*index-map`, which doesn't match your intended pattern ... except that `.` is the regex wildcard so you're actually allowing more patterns we don't want to match. More on that below. 😆